### PR TITLE
Update apr to version 1.5.1

### PR DESCRIPTION
**Pull request by GaetanCambier**
apr version 1.4.6  not availaible on web site

### CODE REVIEW

File: cross/apr/Makefile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 PKG_NAME = apr
-PKG_VERS = 1.4.6
+PKG_VERS = 1.5.1
 PKG_EXT = tar.bz2
 PKG_DIST_NAME = $(PKG_NAME)-$(PKG_VERS).$(PKG_EXT)
-PKG_DIST_SITE = http://apache.crihan.fr/dist//apr
+PKG_DIST_SITE = http://www.apache.org/dist/apr/
```
Comment (Dr-Bean) [inline cross/apr/Makefile]: No trailing slash here

**Comment by Dr-Bean** [comment]
Haven't verified, but seems weird to me that it needs a native build, I would think a `./configure, make, make install` would do it with the relevant flags. Otherwise, looks okay.

**Comment by GaetanCambier** [comment]
the native build is in the pr

**Comment by GaetanCambier** [comment]
done

**Comment by Dr-Bean** [comment]
Might be something on my side, but this isn't working for me. The compile errors out with `bin/bash: tools/gen_test_char_native: cannot execute binary file`. Can you recheck that it's successful on your side after you clean both the native and cross builds?

On that note, how about dropping the native build, and patching out the tools/gen_test_char check altogether? See here for an example: http://lists.openembedded.org/pipermail/openembedded-core/2014-June/093158.html

**Comment by Dr-Bean** [comment]
Ah, figured it out. The native target has to be built first, can't build it as part of the cross make process: probably Make- or environment flags being passed on causing that to happen.
Maybe that can be fixed, either via my previous suggestion, or by correcting the flags in the native's Makefile.

**Comment by GaetanCambier** [comment]
i have compared compile.log from native and cross pakage and the make grogram used is not correct when cross compil the native package. i test a correction

**Comment by Diaoul** [comment]
On that topic, I think we can clear env variables when buiding a native package.

**Comment by GaetanCambier** [comment]
resolbed by compiling all native package

**Pull request closed without merging**